Automattic/jetpack
# Social: Replace + icon with "Manage connections" button in editor

The + icon beside the connection list does more than adding a connection, thus we should replace that + icon with a button/link that says "Manage connections"

Related p1717676583920509-slack-C02JJ910CNL

## Proposed changes:

* Replace the + icon with connections management link/button
* Display the connections list with names and icons and a toggle

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






<details>
    <summary>To simulate a broken connection, you can make these changes (click to expand)</summary>

```diff
diff --git a/projects/packages/publicize/src/class-publicize.php b/projects/packages/publicize/src/class-publicize.php
index 9b52ce51f5..cb3234df0a 100644
--- a/projects/packages/publicize/src/class-publicize.php
+++ b/projects/packages/publicize/src/class-publicize.php
@@ -271,8 +271,10 @@ class Publicize extends Publicize_Base {
                $connection_results     = json_decode( wp_remote_retrieve_body( $response ), true );
                $connection_results_map = array();

+               $count = 0;
+
                foreach ( $connection_results as $connection_result ) {
-                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'ok' : 'broken';
+                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] && ( $count++ ) ? 'ok' : 'broken';
                }
                foreach ( $connections as $key => $connection ) {
                        if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
```
</details>

* Create a new post open Jetpack/Social sidebar
* Under "Share this post", confirm that you see the connections list as in the design
* Confirm that you see the "Manage connections" link below the connections list
* Click on it to confirm that it opens the connections modal
* Test the above for WPCOM sites
* Confirm that the "Manage connections" link points to Calypso

New post

<img width="291" alt="Screenshot 2024-06-19 at 6 07 28 PM" src="https://github.com/Automattic/jetpack/assets/18226415/01342efe-657b-4e13-a012-a042adff04d6">

Existing post
<img width="299" alt="Screenshot 2024-06-19 at 5 27 56 PM" src="https://github.com/Automattic/jetpack/assets/18226415/ebdfe526-61f7-49ed-aaae-fb918cc2b9ff">
